### ISSUE TITLE
This socket has been ended by the other party

### ISSUE THREAD

**Issue by kryptamine** [cypress-io/cypress#5787]
### Current behavior:
Sometimes in CI pipeline, tests are failing with error
```
{ Error: This socket has been ended by the other party
    at TLSSocket.writeAfterFIN [as write] (net.js:404:14)
    at ClientRequest.<anonymous> (/root/.cache/Cypress/3.6.1/Cypress/resources/app/packages/server/node_modules/http-proxy/lib/http-proxy/passes/ws-incoming.js:115:16)
    at ClientRequest.emit (events.js:194:13)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:564:23)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:113:17)
    at TLSSocket.socketOnData (_http_client.js:451:22)
    at TLSSocket.emit (events.js:194:13)
    at addChunk (_stream_readable.js:296:12)
    at readableAddChunk (_stream_readable.js:277:11)
    at TLSSocket.Readable.push (_stream_readable.js:232:10)
    at TLSWrap.onStreamRead (internal/stream_base_commons.js:165:17)
 code: 'EPIPE' }
```
But the exit code is 0 and cypress creates file `core` in the root directory, having root permissions, and we can't delete it automatically from jenkins for example because of wrong permissions
### Desired behavior:

exit code 1 and core inside root directory does not exist or has permissions to write for other users to have an ability to remove it

### Steps to reproduce: (app code and test code)
Run cypress with: ` docker run --rm -i -v $WORKSPACE/frontend:/e2e -w /e2e cypress/included:3.6.1 --config baseUrl=http://localhost --env apiUrl=http://localhost:4000/api --key somekey`

### Versions
Cypress 3.6.1
Ubuntu 18.04

tnx.

**Comment by pouchyRT10** [comment]
I'm getting the same error on both 3.6.1 and 3.8.0

**Comment by jennifer-shehane** [comment]
Is there a previous version this error does not display within? That would help track down the issue. 

Does this happen on every run? 

Unfortunately we'll have to close this issue if no reproducible example is provided - we cannot run the provided repro without the app to host for example. Can anyone provide a way to reproduce this?

**Comment by pouchyRT10** [comment]
> Does this happen on every run?
Yes, every run in Chrome 79.

Does not occur at all in Electron 78

**Comment by pouchyRT10** [comment]
We cleaned up the test scripts; removed a lot of comments. The error is no longer presenting.

**Comment by jevasco** [comment]
@pouchyRT10 Can you please elaborate on how you were able to resolve this?
1. What is the actual issue? i.e. too much comments?
2. What eventually you did that resolve the issue?

Thanks and appreciate your answer

**Comment by CoryDanielson** [comment]
I'm experiencing the same problem in the latest version of Cypress, too.

I experience the issue while running Cypress against my local development code that is using browser sync. When I deploy my built code somewhere where browser sync will not be used, I do not experience the issue.

**Comment by CoryDanielson** [comment]
This seems to be a duplicate of #6458

**Comment by CoryDanielson** [comment]
Welp the tests actually just passed locally twice in a row (first 2 times they have passed). I'm having a hard time getting this down to a reproducible scenario.

**Comment by misterx** [comment]
Issue is related to https://github.com/http-party/node-http-proxy/issues/1342 . 
This fix solve issue https://github.com/caosm/node-http-proxy/commit/46d02be9533cc800e32b0df510149ca35e7453cb

**Comment by 4knort** [comment]
> Issue is related to [http-party/node-http-proxy#1342](https://github.com/http-party/node-http-proxy/issues/1342) .
> This fix solve issue [caosm/node-http-proxy@46d02be](https://github.com/caosm/node-http-proxy/commit/46d02be9533cc800e32b0df510149ca35e7453cb)

how it can help with cypress ?:)

**Comment by jennifer-shehane** [comment]
Duplicate of https://github.com/cypress-io/cypress/issues/6458 which has a fixed going out in our next release.
